src/warnet/bitcoin.py

```diff
@@ -23,23 +24,24 @@ def bitcoin():
 @click.argument("tank", type=str)
 @click.argument("method", type=str)
 @click.argument("params", type=str, nargs=-1)  # this will capture all remaining arguments
-def rpc(tank: str, method: str, params: str):
+@click.option("--namespace", default=None, show_default=True)
+def rpc(tank: str, method: str, params: str, namespace: Optional[str]):
     """
     Call bitcoin-cli <method> [params] on <tank pod name>
     """
     try:
-        result = _rpc(tank, method, params)
+        result = _rpc(tank, method, params, namespace)
     except Exception as e:
         print(f"{e}")
         sys.exit(1)
     print(result)
 
 
-def _rpc(tank: str, method: str, params: str):
+def _rpc(tank: str, method: str, params: str, namespace: Optional[str] = None):
     # bitcoin-cli should be able to read bitcoin.conf inside the container
     # so no extra args like port, chain, username or password are needed
-    namespace = get_default_namespace()
-
+    if not namespace:
+        namespace = get_default_namespace()
```

Comment:
i wonder if the repetition of this can be smoothed out with a function decorator like `@needs_namespace` that can be applied to all relevant commands

Reply:
Fixed here: https://github.com/bitcoin-dev-project/warnet/pull/616/commits/2a906e28b5ed17efe74a75976c4781507b501af4